packages/rum/run-script.js

```diff
@@ -139,37 +139,37 @@ function runSauceTests(serve = 'true') {
   if (serve === 'true') {
     servers = serveE2e('./', 8000)
   }
+  /**
+   * Decides the saucelabs test status
+   */
+  let exitCode = 0
+  const loggerOpts = {
+    stdout: process.stdout,
+    stderr: process.stderr
+  }
   /**
    * Since there is no easy way to reuse the sauce connect tunnel even using same tunnel identifier,
    * we launch the sauce connect tunnel before starting all the saucelab tests
    *
    * Unit tests are not run in parallel with E2E because of concurrency limit in saucelabs
    */
   const sauceConnectOpts = getSauceConnectOptions()
-  testUtils.runSauceConnect(sauceConnectOpts, () => {
-    runAll('test:unit', {
-      stdout: process.stdout,
-      stderr: process.stderr
-    })
-      .then(() =>
-        runAll(['test:e2e:supported', 'test:e2e:failsafe'], {
-          parallel: true,
-          aggregateOutput: true,
-          printLabel: true,
-          stdout: process.stdout,
-          stderr: process.stderr
-        })
-      )
-      .then(() => {
-        console.log('All Unit and E2E Sauce Tests done')
-      })
-      .catch(err => {
-        console.log('Sauce Tests Failed', err)
-      })
-      .then(() => {
-        servers.map(s => s.close())
-        process.exit(0)
+  testUtils.runSauceConnect(sauceConnectOpts, async sauceConnectProcess => {
+    try {
+      await runAll('test:unit', loggerOpts)
+      await runAll(['test:e2e:supported', 'test:e2e:failsafe'], {
+        parallel: true,
```

Comment:
Will this make the saucelab to timeout more often, since we could surpass the concurrency limit?

Reply:
No it does not, I was trying to run unit tests parallel that was the issue i mentioned yesterday.

Reply:
But this still runs 'test:e2e:supported' and 'test:e2e:failsafe' in parallel right?

Reply:
yes this is still run in parallel and does no harm to the timeouts  as far i tested it.. If we include deprecate supportfor more browsers in future, we might hit the limit and would have to run it in series for the time being.